packages/react/src/PatientSummary/SexualOrientation.tsx

```diff
@@ -0,0 +1,141 @@
+  readonly patient: Patient;
+  readonly encounter?: Encounter;
+  readonly sexualOrientation?: Observation;
+}
+
+export function SexualOrientation(props: SexualOrientationProps): JSX.Element {
+  const medplum = useMedplum();
+  const { patient, encounter } = props;
+  const [sexualOrientation, setSexualOrientation] = useState<Observation | undefined>(props.sexualOrientation);
+  const [opened, { open, close }] = useDisclosure(false);
+
+  const handleSubmit = useCallback(
+    (formData: Record<string, string>) => {
+      const code = formData.sexualOrientation as SexualOrientationCode;
+      medplum
+        .createResource<Observation>({
+          resourceType: 'Observation',
+          meta: {
+            profile: [HTTP_HL7_ORG + '/fhir/us/core/ValueSet/us-core-sexual-orientation'],
+          },
+          status: 'final',
+          category: [
+            {
+              coding: [
+                {
+                  system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/observation-category',
+                  code: 'social-history',
+                  display: 'Social History',
+                },
+              ],
+              text: 'Social History',
+            },
+          ],
+          code: {
+            coding: [
+              {
+                system: LOINC,
+                code: '76690-7',
+                display: 'Sexual orientation',
+              },
+            ],
+            text: 'Sexual orientation',
+          },
+          subject: createReference(patient),
+          encounter: encounter ? createReference(encounter) : undefined,
+          effectiveDateTime: new Date().toISOString(),
+          valueCodeableConcept: {
+            coding: [
+              {
+                system: CodesToSystem[code],
+                code: formData.sexualOrientation,
+              },
+            ],
+            text: CodesToText[code],
+          },
+        })
+        .then((newSexualOrientation) => {
+          setSexualOrientation(newSexualOrientation);
+          close();
+        })
+        .catch(console.error);
+    },
+    [medplum, patient, encounter, close]
+  );
+
+  return (
+    <>
+      <Group justify="space-between">
+        <Text fz="md" fw={700}>
+          Sexual Orientation
+        </Text>
+        <Anchor
+          href="#"
+          onClick={(e) => {
+            killEvent(e);
+            open();
+          }}
+        >
+          + Edit
+        </Anchor>
+      </Group>
+      {sexualOrientation?.valueCodeableConcept ? (
+        <Box>
+          <Badge variant="light">
+            <CodeableConceptDisplay value={sexualOrientation.valueCodeableConcept} />
+          </Badge>
+        </Box>
+      ) : (
+        <Text>(none)</Text>
+      )}
+      <Modal opened={opened} onClose={close} title="Set Sexual Orientation">
+        <Form onSubmit={handleSubmit}>
+          <Stack>
+            <Radio.Group name="sexualOrientation" label="Sexual Orientation" required>
```

Comment:
🤔 

I wonder if we should have a "radio button" option on `<CodeableConceptInput>`?  Most of the logic would be the same, just a different presentation mode.  That would have saved a lot of code here.